client/state/data-layer/wpcom-http/utils.js

```diff
@@ -66,28 +70,42 @@ export const getProgress = action => get( action, 'meta.dataLayer.progress', nul
  *   onError    :: ReduxStore -> Action -> Dispatcher -> ErrorData
  *   onProgress :: ReduxStore -> Action -> Dispatcher -> ProgressData
  *
- * @param {Function} initiator called if action lacks response meta; should create HTTP request
- * @param {Function} onSuccess called if the action meta includes response data
- * @param {Function} onError called if the action meta includes error data
+ * @param {Function} initiator    called if action lacks response meta; should create HTTP request
+ * @param {Function} onSuccess    called if the action meta includes response data
+ * @param {Function} onError      called if the action meta includes error data
  * @param {Function} [onProgress] called on progress events when uploading. The default
  *                                behavior of this optional handler is to do nothing.
+ * @param {Object} options        Options object
+
  * @returns {?*} please ignore return values, they are undefined
  */
-export const dispatchRequest = ( initiator, onSuccess, onError, onProgress = noop ) => ( store, action, next ) => {
-	const error = getError( action );
-	if ( error ) {
-		return onError( store, action, next, error );
-	}
 
-	const data = getData( action );
-	if ( data ) {
-		return onSuccess( store, action, next, data );
+export const dispatchRequest = ( initiator, onSuccess, onError, onProgress = noop, options = {} ) => {
+	const { freshness } = options;
+	let initiatorToUse, onSuccessToUse;
+	if ( freshness ) {
```

Comment:
i know it was suggested by @dmsnell, but this higher-order-initiator approach doesn't strike me as particularly clean